de.metas.business.rest-api-impl/src/test/java/de/metas/rest_api/invoicecandidates/impl/InvoiceCandidatesRestControllerImplTest.java

```diff
@@ -0,0 +1,59 @@
+package de.metas.rest_api.invoicecandidates.impl;
+
+import static org.adempiere.model.InterfaceWrapperHelper.newInstance;
+import static org.adempiere.model.InterfaceWrapperHelper.saveRecord;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.adempiere.test.AdempiereTestHelper;
+import org.compiere.model.IQuery;
+import org.junit.Before;
+import org.junit.Test;
+
+import de.metas.invoicecandidate.InvoiceCandidateId;
+import de.metas.invoicecandidate.model.I_C_Invoice_Candidate;
+import de.metas.process.PInstanceId;
+import de.metas.rest_api.invoicecandidates.request.JsonInvoiceCandidate;
+import de.metas.util.rest.ExternalId;
+
+public class InvoiceCandidatesRestControllerImplTest {
```

Comment:
Generally, please check out https://phauer.com/2019/modern-best-practices-testing-java/ (pasted into our dev-channel be @teosarca a few weeks ago)

In particular: 
* please be more specific, than ".isGreaterThan(0);" and "isNotEmpty()" after all you controll exactly how many ICs there are to be returned
* also, please add some more tests
  * empty list of ExternalLineIds
  * non-empty list of ExternalLineIds
* also add more invoice candidates and veryfiy that nose ICs that are not mathced shall *not* selected. For all I know from this test, the code might return all IC-records, or a random-non-empty subset of all IC-records.

Reply:
done.